conan/tools/cmake/presets.py

```diff
@@ -100,7 +100,8 @@ def _configure_preset(conanfile, generator, cache_variables, toolchain_file, mul
         if preset_prefix:
             name = f"{preset_prefix}-{name}"
         if not multiconfig and build_type:
-            cache_variables["CMAKE_BUILD_TYPE"] = build_type
+            if not "CMAKE_BUILD_TYPE" in cache_variables:
```

Comment:
This wouldnt be a fix for the test, because the test doesn't use the presets at all, seems unrelated. It is true that command line usage of ``cmake --preset`` might require other changes that my PR didn't consider, but for the test below, it is unrelated.